pkg/kubelet/cm/dra/manager.go

```diff
@@ -185,20 +186,20 @@ func (m *ManagerImpl) PrepareResources(pod *v1.Pod) error {
 				pod.Name, pod.UID, *claimName, resourceClaim.UID)
 		}
 
-		// If no container actually uses the claim, then we don't need
-		// to prepare it.
-		if !claimIsUsedByPod(podClaim, pod) {
-			klog.V(5).InfoS("Skipping unused resource", "claim", claimName, "pod", pod.Name)
-			continue
-		}
-
```

Comment:
Note to self: This is because network devices should still be prepared even if no container consumes them directly.

Reply:
Right. Presumably such a driver then will return no CDI device IDs, because nothing would be done with them anyway.

Where it gets fuzzy for me is how such a driver then influences the container and/or network creation.

Reply:
My understanding is -- once this is done -- an external client would read the list of CDIDevices from the PodResources API and do something with it. More specifically, a CNI plugin would do this.

/cc @moshe010 for more info

Reply:
yes that right, multus already has the support for it
https://github.com/k8snetworkplumbingwg/multus-cni/pull/1078

Reply:
How did that work in the past when NodePrepareResources wasn't called? Was something reported in the PodResource API *without* calling it?

See also https://github.com/kubernetes/kubernetes/pull/125488#discussion_r1665913973

Reply:
No. I'm assuming a `pause` container (or equivalent) must have been run to consume the claim.